### PR TITLE
NETOBSERV-489: Use authorize feature from console plugin API

### PR DESCRIPTION
**Pull request by OlivierCazade**
This PR enable the new option from the console from this PR:

https://github.com/netobserv/network-observability-console-plugin/pull/198

### CODE REVIEW

File: api/v1alpha1/flowcollector_types.go [hunk 1]
```diff
@@ -446,6 +446,11 @@ type FlowCollectorConsolePlugin struct {
 	// imagePullPolicy is the Kubernetes pull policy for the image defined above
 	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
 
+	//+kubebuilder:default:=false
+	// sendAuthToken is a flag to enable or disable forwarind auth token
+	// if enabled, this overide loki.sendAuthToken
+	ForwardUserAuthToken bool `json:"forwardUserAuthToken,omitempty"`
```
Comment (mariomac) [inline api/v1alpha1/flowcollector_types.go:452]: Following the API recommendations, discouraging the use of bools, maybe I'd move this to a string property. Something like: `userAuthToken: {BLOCK | FORWARD}`

Comment (jpinsonneau) [reply api/v1alpha1/flowcollector_types.go:452]: maybe we should merge with `FlowCollectorLoki.SendAuthToken` to a single enum then ? :thinking:

Comment (OlivierCazade) [reply api/v1alpha1/flowcollector_types.go]: I change the value to an enum.

Comment (OlivierCazade) [reply api/v1alpha1/flowcollector_types.go]: > maybe we should merge with `FlowCollectorLoki.SendAuthToken` to a single enum then ? thinking

`FlowCollectorLoki.SendAuthToken` apply to both FLP and Console plugin while this new value only apply to the Console Plugin.

Also `FlowCollectorLoki.SendAuthToken` is still required for FLP when the user token forwarding is enabled.

Comment (jpinsonneau) [reply api/v1alpha1/flowcollector_types.go]: Yes. My thinking was about having `FlowCollectorLoki.SendAuthToken` as:
```yaml
enum:
  - IGNORE
  - SERVICE_ACCOUNTS
  - MIXED
  type: string
```

With proper documentation to explain `MIXED` behavior

Comment (OlivierCazade) [reply api/v1alpha1/flowcollector_types.go]: Done

**Review by mariomac — APPROVED** [review]
LGTM. I haven't tested it.

**Pull request merged**